Src/Support/GoogleApis.Core/Apis/Requests/Parameters/ParameterUtils.cs

```diff
@@ -121,7 +124,25 @@ private static void IterateParameters(object request, Action<RequestParameterTyp
                 // Call action with the type name and value.
                 if (propertyType.IsValueType || value != null)
                 {
-                    action(attribute.Type, name, value);
+                    if (attribute.Type == RequestParameterType.UserDefinedQueries)
+                    {
+                        if (typeof(IEnumerable<KeyValuePair<string, string>>).IsAssignableFrom(value.GetType()))

```

Comment:
Simply checking if ```value.GetType() == IEnumerable<KeyValuePair<string, string>>``` didn't work for other generic collections like List, though it did seem to work for an Array (which I had originally in the unit test). using the ```IsAssignableFrom()``` method seemed to have done the trick in making this evaluation more accurate / flexible.